### PR TITLE
Supported update of json during eval

### PR DESCRIPTION
**Pull request by pfav**
This changes add the support of transforming the json (nested structure) during the evaluation.
Also some minor feature where added (Recursive descent, Python-slicing, ...)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,28 +15,51 @@ Examples
...
+ejsonpath:tr("$.items[?(@.id == 0)].value", O, fun(_) -> xxx end).
+{
+  #{<<"items">> =>
+       [#{<<"id">> => 0,<<"value">> => xxx},
+        #{<<"id">> => 1,<<"value">> => yyy}]},
+  ["$['items'][0]['value']"]
+ 
+
+% delete item if query match
+ejsonpath_transform:tr("$.items[?(@.id == 0)]", O, fun (_) -> delete end, #{}, []))
```
Comment (seriyps) [inline README.md]: and would be nice to explain what are the last 2 arguments for?

And other possible return values for `TransformFun`:
```
{error, Error}
{ok, Result}
```

**Comment by pfav** [comment]
Hi!
thanks for the notes. I've made the updates you suggested. hope that this changes are useful

**Comment by seriyps** [comment]
Looks good for me. But I don't have commit rights to this repo anymore (left Ostrovok long ago), so, maybe someone from @ostrovok-team could merge this?

**Comment by vanadium23** [comment]
Merged, thank you for contribution :tada:

**Comment by seriyps** [comment]
Thanks! Pushed to hex.pm https://hex.pm/packages/ejsonpath/0.2.1

@vanadium23 would you mind to tag current master as `0.2.1`?

```
git pull origin master
git tag 0.2.1
git push origin --tags
```

**Comment by vanadium23** [comment]
@seriyps done

**Pull request merged**